src/workspaces/SettingsModal/BucketLifecycleSettings.tsx

```diff
@@ -0,0 +1,122 @@
+import { CreatableSelect, ExternalLink, Switch, useUniqueId } from '@terra-ui-packages/components';
+import _ from 'lodash/fp';
+import React, { ReactNode } from 'react';
+import { NumberInput } from 'src/components/input';
+import { FormLabel } from 'src/libs/forms';
+import { suggestedPrefixes } from 'src/workspaces/SettingsModal/utils';
+
+interface BucketLifecycleSettingsProps {
+  lifecycleRulesEnabled: boolean;
+  setLifecycleRulesEnabled: (enabled: boolean) => void;
+  lifecycleAge: number | null;
+  setLifecycleAge: (age: number | null) => void;
+  prefixes: string[];
+  setPrefixes: (prefixes: string[]) => void;
+  isOwner: boolean;
+}
+
+const BucketLifecycleSettings = (props: BucketLifecycleSettingsProps): ReactNode => {
+  const {
+    lifecycleRulesEnabled,
+    setLifecycleRulesEnabled,
+    lifecycleAge,
+    setLifecycleAge,
+    prefixes,
+    setPrefixes,
+    isOwner,
+  } = props;
+
+  const switchId = useUniqueId('switch');
+  const daysId = useUniqueId('days');
+  const descriptionId = useUniqueId('description');
+
+  const prefixOptions = () => {
+    // Append together suggested options and any options the user has already selected.
+    const allOptions = _.uniq(_.concat(_.values(suggestedPrefixes), prefixes));
+    return _.map((value) => ({ value, label: value }), allOptions);
+  };
+
+  return (
+    <>
+      <div style={{ display: 'flex', flexDirection: 'row', alignItems: 'center', marginTop: '.75rem' }}>
+        <FormLabel
+          htmlFor={switchId}
+          style={{ fontWeight: 600, whiteSpace: 'nowrap', marginRight: '.5rem', marginTop: '.5rem' }}
+        >
+          Lifecycle Rules:
+        </FormLabel>
+        <Switch
+          onLabel=''
+          offLabel=''
+          onChange={(checked: boolean) => {
+            setLifecycleRulesEnabled(checked);
+            if (!checked) {
+              // Clear out the values being display to reduce confusion
+              setLifecycleAge(null);
+              setPrefixes([]);
+            }
+          }}
+          id={switchId}
+          checked={lifecycleRulesEnabled}
+          width={40}
+          height={20}
+          aria-describedby={descriptionId}
+          disabled={!isOwner}
```

Comment:
In addition to disabling the switch if you aren't an owner, I need to disable prefix select and date input field (otherwise readers/writers can modify the values in the modal, though they can't save them because the Save button is disabled).

I will fold this change into my Soft Delete PR, as it will be ready soon.